ludwig/schema/model_types/base.py

```diff
@@ -88,12 +91,17 @@ def from_dict(config: ModelConfigDict) -> "ModelConfig":
         if isinstance(backend, str):
             config[BACKEND] = {"type": backend}
 
+        # JSON schema validation.
+        check_schema(config)
+
         cls = model_type_schema_registry[model_type]
         schema = cls.get_class_schema()()
         try:
             config_obj: ModelConfig = schema.load(config)
         except ValidationError as e:
             raise ConfigValidationError(f"Config validation error raised during config deserialization: {e}") from e
+
+        get_config_check_registry().check_config(config_obj)
```

Comment:
Please move this to `__post_init__` above.

Reply:
Done.